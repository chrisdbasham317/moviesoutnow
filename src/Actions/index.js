export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (msg) => ({
  type: 'HAS_ERRORED',
  msg
})

export const setRecentMovies = (movies) => ({
  type: 'SET_RECENT_MOVIES',
  movies
})

export const setGenres = (genres) => ({
  type: 'SET_GENRES',
  genres
})

export const currentUser = currentUser => ({
  type: 'CURRENT_USER',
  currentUser
})

export const toggleModal = toggleState => ({
  type: 'TOGGLE_MODAL',
  toggleState
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const setFavorites = (favorites) => ({
  type: 'SET_FAVORTIES',
  favorites
})