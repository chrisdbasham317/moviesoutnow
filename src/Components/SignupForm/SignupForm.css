.SignupForm {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  width: 400px;
  color: #FFF;
}

.signup-login {
  display: flex;
  justify-content: space-between;
  margin-bottom: 20px;
}

.first--last--name__div{
  display: flex;
  justify-content: space-between;
}

.name__div--input,
.email--password--SU__div {
  display: flex;
  flex-direction: column;
}

.name__div--input {
  width: 45%;
}

.signup-text {
  margin: 0;
  color: #74FBFF;

}

.signup--login--button {
  border: none;
  background: none;
  color: #FFF;
  cursor: pointer;
}

.signup--submit__button,
.email--password--SU__input,
.signup--name--input {
  border: none;
  height: 45px;
  border-radius: 5px;
  padding-left: 10px;;
  font-size: 20px;
  margin-top: 8px;
  margin-bottom: 20px;
}

.signup--submit__button {
  padding-left: 0;
  margin-bottom: 0;
  width: 45%;
  background-color: #74FBFF;
  font-weight: 500;
  cursor: pointer;
}

.signup--submit__button:disabled {
  cursor: default;
}