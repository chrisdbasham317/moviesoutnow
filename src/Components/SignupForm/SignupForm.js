import React from "react"
import "./SignupForm.css"

export class SignupForm extends React.Component {
  constructor() {
    super()
    this.state = {
      firstName: "",
      lastName: "",
      email: "",
      password: ""
    }
  }

render() {
  return(
    <form className="SignupForm">
      <div className="signup-login">
        <h3 className="signup-text">SIGN UP</h3>
        <button type="button" className="signup--login--button">Current User Login</button>
      </div>
      <div className="first--last--name__div">
        <div className="name__div--input">
          <label
          className="signup--label"
          htmlFor='firstNameSU'
          >First Name</label>
          <input
          type="text"
          className="signup--name--input"
          id="signup--name--input"
          placeholder="First Name"
          name="firstName"
          value={this.state.firstName}
          />
        </div>
        <div className="name__div--input">
          <label
          className="signup--label"
          htmlFor='lastNameSU'
          >Last Name</label>
          <input
          type="text"
          className="signup--name--input"
          id="signup--name--input"
          placeholder="Last Name"
          name="lastName"
          value={this.state.lastName}
          />
        </div>
      </div>
      <div>
        <div className="email--password--SU__div">
          <label
          className="email--password--SU__label"
          htmlFor="email">Email</label>
          <input
          type="text"
          className="email--password--SU__input"
          id="email--SU"
          placeholder="Email"
          name="email"
          value={this.state.email}
          />
        </div>
        <div className="email--password--SU__div">
          <label
          className="email--password--SU__label"
          htmlFor="password">Passowrd</label>
          <input
          type="text"
          className="email--password--SU__input"
          id="password--SU"
          placeholder="Password"
          name="password"
          value={this.state.password}
          />
        </div>
      </div>
      <button
      type="button"
      className="signup--submit__button">SIGN UP</button>
    </form>

  )
}

}

export default SignupForm
