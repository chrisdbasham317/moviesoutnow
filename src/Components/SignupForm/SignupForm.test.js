import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { toggleModal, hasErrored } from '../../Actions';
import { SignupForm } from './SignupForm';

describe("SignupForm", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <SignupForm
        toggleModal={true}
        error={''}
        toggleModal={jest.fn()}
        postNewUser={jest.fn()}
        hasErrored={jest.fn()}
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})