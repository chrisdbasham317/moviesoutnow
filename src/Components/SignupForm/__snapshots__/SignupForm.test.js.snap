// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm should match the snapshot 1`] = `
<form
  className="SignupForm"
>
  <div
    className="signup-login"
  >
    <h3
      className="signup-text"
    >
      SIGN UP
    </h3>
    <img
      alt="Close Button"
      className="signup_close_button"
      onClick={[Function]}
      src="close_btn.png"
    />
  </div>
  <div
    className="first--last--name__div"
  >
    <div
      className="name__div--input"
    >
      <label
        className="signup--label"
        htmlFor="firstNameSU"
      >
        First Name
      </label>
      <input
        className="signup--name--input"
        id="signup--name--input fname"
        name="firstName"
        onChange={[Function]}
        placeholder="First Name"
        type="text"
        value=""
      />
    </div>
    <div
      className="name__div--input"
    >
      <label
        className="signup--label"
        htmlFor="lastNameSU"
      >
        Last Name
      </label>
      <input
        className="signup--name--input"
        id="signup--name--input lname"
        name="lastName"
        onChange={[Function]}
        placeholder="Last Name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <div
      className="email--password--SU__div"
    >
      <label
        className="email--password--SU__label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="email--password--SU__input"
        id="email--SU"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
    </div>
    <div
      className="email--password--SU__div"
    >
      <label
        className="email--password--SU__label"
        htmlFor="password"
      >
        Passowrd
      </label>
      <input
        className="email--password--SU__input"
        id="password--SU"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    className="signup--submit__button"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    SIGN UP
  </button>
  <div
    className="error--message--spacing"
  >
    <div
      className="signup__error--none"
    >
      <img
        alt="Error Icon"
        className="error__icon"
        src="error_icon.png"
      />
      <p
        className="signup__error__p"
      >
        Account with this email already exisits.
      </p>
    </div>
  </div>
</form>
`;
